src/test/scala/is/hail/annotations/StagedDecoderSuite.scala

```diff
@@ -0,0 +1,217 @@
+package is.hail.annotations
+
+import is.hail.SparkSuite
+import is.hail.expr._
+import is.hail.utils._
+import org.testng.annotations.{BeforeMethod, Test}
+import is.hail.annotations.StagedDecoder._
+import is.hail.io._
+import is.hail.TestUtils._
+
+class StagedDecoderSuite extends SparkSuite {
```

Comment:
These are good but a Gen-based random decoder test like UnsafeSuite.testCodec would be awesome.